misc/pcap.c

```diff
@@ -1327,3 +1327,54 @@ get_iface_from_ip (const char *local_ip)
 
   return if_name;
 }
+
+/** @brief Get the interface index depending on the target's IP
+ *
+ * @param[in] ipaddr The ip address of the target.
+ * @param[out] ifindex the index of the selected iface
+ *
+ * @return 0 on success, otherwise -1.
+ */
+int
+get_iface_index (struct in6_addr *ipaddr, int *ifindex)
+{
+  struct in6_addr src_addr;
+  struct ifreq ifr;
+  char *if_name, *ip_address;
+  int soc;
+
+  // We get the local address to use, with the remote address.
+  memset (&src_addr, '\0', sizeof (struct in6_addr));
+  v6_getsourceip (&src_addr, ipaddr);
+  ip_address = addr6_as_str (&src_addr);
+
+  // Once with the local ip address, we get the source iface name
+  if_name = get_iface_from_ip (ip_address);
+  g_free (ip_address);
+  if (!if_name)
+    {
+      g_debug ("%s: Missing interface name", __func__);
+      return -1;
+    }
+
+  // Create the raw socket
+  soc = socket (AF_INET, SOCK_DGRAM, 0);
+  if (soc == -1)
+    {
+      g_debug ("%s: %s", __func__, strerror (errno));
```

Comment:
`#include <errno.h>` seems to be missing.

Reply:
Solved with https://github.com/greenbone/openvas-scanner/pull/954/commits/26594e4409a27906dc11393ea940a96bd1aedf8a

Reply:
I think its also needed for pcap.c

Reply:
It is still not compiling

Reply:
why is it passing the checks, is there anything wrong with our CI? Also, in my local env is compiling without problems.

Reply:
Anyway, I added it with 5c4c45f3f58355c6fe0c85cdd18163a231e0329a